pkg/utils/utils.go

```diff
@@ -3,164 +3,123 @@ package utils
 	return bytes, err
 }
 
-// Generate a cryptographically secure random string.
-// Returned string is encoded using [encoding.RawURLEncoding]
-// which makes it safe to use in URLs and file names.
+// RandomString generates a cryptographically secure random string.
+// The returned string is encoded using base64.RawURLEncoding, making it safe for URLs and file names.
 func RandomString(length int) (string, error) {
 	encoding := base64.RawURLEncoding
-	// each byte (8 bits) gives us 4/3 base64 (6 bits) characters,
-	// we account for that conversion and add one to handle truncation
-	b64size := encoding.DecodedLen(length) + 1
+	b64size := encoding.DecodedLen(length) + 1 // Account for base64 encoding overhead
 	randomBytes, err := RandomBytes(b64size)
 	if err != nil {
 		return "", err
 	}
-	// trim down to the original requested size since we added one above
 	return encoding.EncodeToString(randomBytes)[:length], nil
 }
 
-// buildCSPDirectives takes the content security policy configuration from the server and constructs
-// a complete set of directives for the Content-Security-Policy-Report-Only header.
-// The constructed directives will include the default sources and the supplied configuration.
-
-func BuildCSPDirectives(k8sMode, pluginsCSP, indexPageScriptNonce string) ([]string, error) {
+// BuildCSPDirectives constructs a complete set of Content Security Policy (CSP) directives
+// based on the provided configuration and mode.
+func BuildCSPDirectives(k8sMode string, pluginsCSP serverconfig.MultiKeyValue, indexPageScriptNonce string) ([]string, error) {
 	nonce := fmt.Sprintf("'nonce-%s'", indexPageScriptNonce)
 
-	// The default sources are the sources that are allowed for all directives.
+	// Initialize directives with default values
 	// When running on-cluster, the default sources are just 'self' and 'console.redhat.com'.
 	// When running off-cluster, 'http://localhost:8080' and 'ws://localhost:8080' are appended to the
 	// default sources. Image source, font source, and style source only use 'self' and
-	// 'http://localhost:8080'.
-	baseUriDirective := []string{baseURI, self}
-	defaultSrcDirective := []string{defaultSrc, self, consoleDot}
-	imgSrcDirective := []string{imgSrc, self}
-	fontSrcDirective := []string{fontSrc, self}
-	scriptSrcDirective := []string{scriptSrc, self, consoleDot}
-	styleSrcDirective := []string{styleSrc, self}
-	if k8sMode == "off-cluster" {
-		baseUriDirective = append(baseUriDirective, []string{httpLocalHost, wsLocalHost}...)
-		defaultSrcDirective = append(defaultSrcDirective, []string{httpLocalHost, wsLocalHost}...)
-		imgSrcDirective = append(imgSrcDirective, httpLocalHost)
-		fontSrcDirective = append(fontSrcDirective, httpLocalHost)
-		scriptSrcDirective = append(scriptSrcDirective, []string{httpLocalHost, wsLocalHost}...)
-		styleSrcDirective = append(styleSrcDirective, httpLocalHost)
+	directives := map[string][]string{
+		baseURI:        {self},
+		defaultSrc:     {self, consoleDot},
```

Comment:
Can we get away with only adding `consoleDot` to `scriptSrc` and `connectSrc`? It would be better if we can avoid putting it in `defaultSrc`.